src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/TaintedDataSymbolMapExtensions.cs

```diff
@@ -116,24 +116,26 @@ public static bool IsSourceProperty(this TaintedDataSymbolMap<SourceInfo> source
         }
 
         /// <summary>
-        /// Determines if the given array can be a tainted data source when its elements are all constant.
+        /// Determines if the given array can be a tainted data source.
         /// </summary>
         /// <param name="sourceSymbolMap"></param>
         /// <param name="arrayTypeSymbol"></param>
         /// <returns></returns>
-        public static bool IsSourceConstantArrayOfType(this TaintedDataSymbolMap<SourceInfo> sourceSymbolMap, IArrayTypeSymbol arrayTypeSymbol)
+        public static bool IsSourceArray(this TaintedDataSymbolMap<SourceInfo> sourceSymbolMap, IArrayTypeSymbol arrayTypeSymbol, out TaintArrayKind taintArray)
         {
             if (arrayTypeSymbol.ElementType is INamedTypeSymbol elementType)
             {
                 foreach (SourceInfo sourceInfo in sourceSymbolMap.GetInfosForType(elementType))
                 {
-                    if (sourceInfo.TaintConstantArray)
+                    if (sourceInfo.TaintArray != TaintArrayKind.None)
                     {
-                        return true;
+                        taintArray = sourceInfo.TaintArray;
+                        return taintArray != TaintArrayKind.None;
```

Comment:
>taintArray != TaintArrayKind.None; [](start = 31, length = 34)

Isn't this always true at this point?

What if there are multiple SourceInfos with TaintArray != None from sourceSymbolMap.GetInfosForType(elementType)?

Reply:
All TaintArray should be consistent.
Or we can set it as All > Constant > None.

Reply:
Is there anything that guarantees that all TaintArrays will be consistent?

I'd be fine with All &gt; Constant &gt; None.

Reply:
Looking at how IsSourceArray is used, it seems 'All' just means array arguments. Should 'All' also mean any array instantiation?